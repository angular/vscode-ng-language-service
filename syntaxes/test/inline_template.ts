// SYNTAX TEST "template.ng"
/* clang-format off */

@Component({
//// Property key/value test
  template: '<div></div>',
//^^^^^^^^^                 meta.object-literal.key.ts
//           ^^^^^^^^^^^    template.ng (fake grammar token)

//// String delimiter tests
  template: `<div></div>`,
//          ^               string
//                      ^   string
  template: "<div></div>",
//          ^               string
//                      ^   string
  template: '<div></div>',
//          ^               string
//                      ^   string

//// Parenthesization tests
  template: ( (( '<div></div>' )) ),
//          ^                         meta.brace.round.ts
//            ^^                      meta.brace.round.ts
//               ^                    string
//                           ^        string
//                             ^^     meta.brace.round.ts
//                                ^   meta.brace.round.ts
})
export class TMComponent{}
