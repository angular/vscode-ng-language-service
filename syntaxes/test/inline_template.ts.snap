>// SYNTAX TEST "template.ng"
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ template.ng
>/* clang-format off */
#^^^^^^^^^^^^^^^^^^^^^^^ template.ng
>
>@Component({
#^^^^^^^^^^^^^ template.ng
>//// Property key/value test
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ template.ng
>  template: '<div></div>',
#^^ template.ng
#  ^^^^^^^^ template.ng meta.object-literal.key.ts
#          ^ template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ template.ng
#            ^ template.ng string
#             ^^^^^^^^^^^ template.ng
#                        ^ template.ng string
#                         ^^ template.ng
>
>//// String delimiter tests
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^ template.ng
>  template: `<div></div>`,
#^^ template.ng
#  ^^^^^^^^ template.ng meta.object-literal.key.ts
#          ^ template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ template.ng
#            ^ template.ng string
#             ^^^^^^^^^^^ template.ng
#                        ^ template.ng string
#                         ^^ template.ng
>  template: "<div></div>",
#^^ template.ng
#  ^^^^^^^^ template.ng meta.object-literal.key.ts
#          ^ template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ template.ng
#            ^ template.ng string
#             ^^^^^^^^^^^ template.ng
#                        ^ template.ng string
#                         ^^ template.ng
>  template: '<div></div>',
#^^ template.ng
#  ^^^^^^^^ template.ng meta.object-literal.key.ts
#          ^ template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ template.ng
#            ^ template.ng string
#             ^^^^^^^^^^^ template.ng
#                        ^ template.ng string
#                         ^^ template.ng
>
>//// Parenthesization tests
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^ template.ng
>  template: ( (( '<div></div>' )) ),
#^^ template.ng
#  ^^^^^^^^ template.ng meta.object-literal.key.ts
#          ^ template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ template.ng
#            ^ template.ng meta.brace.round.ts
#             ^ template.ng
#              ^ template.ng meta.brace.round.ts
#               ^ template.ng meta.brace.round.ts
#                ^ template.ng
#                 ^ template.ng string
#                  ^^^^^^^^^^^ template.ng
#                             ^ template.ng string
#                              ^ template.ng
#                               ^ template.ng meta.brace.round.ts
#                                ^ template.ng meta.brace.round.ts
#                                 ^ template.ng
#                                  ^ template.ng meta.brace.round.ts
#                                   ^^ template.ng
>
>//// Comments tests
#^^^^^^^^^^^^^^^^^^^^ template.ng
>  // template: '<div></div>'
#^^^^^ template.ng
#     ^^^^^^^^ template.ng meta.object-literal.key.ts
#             ^ template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#              ^ template.ng
#               ^ template.ng string
#                ^^^^^^^^^^^ template.ng
#                           ^ template.ng string
>  /*
#^^^^^ template.ng
>   * template: '<div></div>'
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ template.ng
>   */
#^^^^^^ template.ng
>  /**
#^^^^^^ template.ng
>   * template: '<div></div>'
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ template.ng
>   */
#^^^^^^ template.ng
>})
#^^^ template.ng
>export class TMComponent{}
#^^^^^^^^^^^^^^^^^^^^^^^^^^^ template.ng
>